src/rasterstats/main.py

```diff
@@ -151,32 +152,68 @@ def gen_zonal_stats(
         features_iter = read_features(vectors, layer)
         for _, feat in enumerate(features_iter):
             geom = shape(feat['geometry'])
-
             if 'Point' in geom.type:
                 geom = boxify_points(geom, rast)
 
-            geom_bounds = tuple(geom.bounds)
+            #Multiple polygons found, and split_multi_geom option is enabled
+            #In this case, we read the polygons in one at a time and concact the raster
+            #values together.  In order to save memory
+            if ('MultiPolygon' in geom.type) and (split_multi_geom):
+                fsrc = None
+                rv_array = None
+                isnodata = None
+                
+                #iterate through individual polygons
+                for singlePolygon in geom.geoms:
```

Comment:
I'd like to see standard PEP8 variable names, i.e. `single_polygon` instead of `singlePolygon` for consistency.

Reply:
Changed variable name